CHANGELOG/v1.7.0.md

```diff
@@ -0,0 +1,433 @@
+## 👌 Kubernetes version support
+
+- Management Cluster: v1.26.x -> v1.29.x
+- Workload Cluster: v1.24.x -> v1.29.x
+
+[More information about version support can be found here](https://cluster-api.sigs.k8s.io/reference/versions.html)
+
+
+## Highlights
+
+* Set MachinePool feature flag default to true + Beta
+* Enable kubeadm feature gates mutation
+* Add public function to create new CRD migrator
+* Add MachineSetReady condition to MachineDeployment
+
+## Changes since v1.6.0
+## :chart_with_upwards_trend: Overview
+- 339 new commits merged
+- 5 breaking changes :warning:
+- 12 feature additions ✨
+- 48 bugs fixed 🐛
+
+## :memo: Proposals
+- MachinePool: Update MachinePool Machines proposal with diagrams (#9664)
+
+## :warning: Breaking Changes
+- API: Remove v1alpha4 API Version (#9939)
+- Dependency: Bump to controller-runtime v0.17.0 & k8s.io v0.29 (#9964)
+- Logging: Make event recorder names consistent for KCP and topology controller (#9862)
+- MachinePool: Set MachinePool feature flag default to true + Beta (#10141)
+- KCP/MachineSet: Objects generated by KCP, MachineSets and MachinePools will now consistently use machine name (#9833)
+
+## :sparkles: New Features
+- API: Enable kubeadm feature gates mutation (#10154)
+- clusterctl: Add k0smotron provider (#9991)
+- clusterctl: Add public function to create new CRD migrator (#10075)
+- Control-plane: Allow mutation of all fields that should be mutable (#9871)
+- MachineDeployment: Add MachineSetReady condition to MachineDeployment (#9262)
+- MachineDeployment: Taint nodes with PreferNoSchedule during rollouts (#10223)
+- MachineHealthCheck: implement annotation to manually mark machines for remediation (#10202)
+- Release: Add dependencies section to release notes tool (#10043)
+- Release: Add release notes expander functionality (#10091)
+- Release: Add utility to create git issues on provider repo (#9110)
+- Testing: Resolve release markers (#9265)
+- Testing: Prepare quickstart, capd and tests for the new release including kind bump (#9872)
+
+## :bug: Bug Fixes
+- API: Use ptr instead of own implementation (#10276)
+- API: Restore v1alpha3/v1alpha4 conversion to fix SSA issue & add e2e test coverage (#10147)
```

Comment:
Let's drop `- API: Remove v1alpha4 API Version (#9939)` above

and add it here instead:

```
- API: Restore v1alpha3/v1alpha4 conversion to fix SSA issue & add e2e test coverage (#9939 #10147) (for more details see #10051)
```